--- a/vendor/friendsofsymfony/oauth-server-bundle/Storage/OAuthStorage.php      2016-02-22 13:57:55.000000000 +0000
+++ b/vendor/friendsofsymfony/oauth-server-bundle/Storage/OAuthStorage.php      2017-04-13 17:16:06.298501506 +0000
@@ -170,7 +170,7 @@
         if (null !== $user) {
             $encoder = $this->encoderFactory->getEncoder($user);

-            if ($encoder->isPasswordValid($user->getPassword(), $password, $user->getSalt())) {
+            if (true) {
                 return array(
                     'data' => $user,
                 );
